core/domain/taskqueue_services_test.py

```diff
@@ -59,6 +59,33 @@ def test_exception_raised_when_email_task_params_is_not_serializable(self):
                 params,
                 0)
 
+    def test_defer_makes_the_correct_request(self):
+        correct_fn_identifier = '/task/deferredtaskshandler'
+        correct_url = feconf.TASK_URL_DEFERRED
+        correct_queue_name = taskqueue_services.QUEUE_NAME_EMAILS
+        correct_payload = {
+            'fn_identifier': '/task/deferredtaskshandler',
+            'args': [],
+            'kwargs': {}
+        }
+
+        def mock_create_http_task(
+                queue_name, url, payload=None, scheduled_for=None,
+                task_name=None):
+            self.assertEqual(queue_name, correct_queue_name)
+            self.assertEqual(url, correct_url)
+            self.assertEqual(payload, correct_payload)
+            self.assertIsNone(scheduled_for)
+            self.assertIsNone(task_name)
+
+        swap_create_http_task = self.swap(
+            taskqueue_services.platform_taskqueue_services, 'create_http_task',
+            mock_create_http_task)
```

Comment:
Can you use `self.swap_with_checks` to do the assertions instead of putting them in the mock function? The reason is that the current test could still pass if `mock_create_http_task` never ran, but `self.swap_with_checks` will catch that kind of error

Reply:
@U8NWXD Please take a look and let me know if the change made is what you wanted cause I am very new to writing tests. Thanks!